File: C#2019/Client/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace WindowsFormsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void axVideoChatSender1_SendStream(object sender, AxVideoChatSenderLib._DvideochatEvents_SendStreamEvent e)
        {
            lblvideobitrate.Text = e.fVideoBitrate.ToString();
            lblaudiobitrate.Text = e.fAudioBitrate.ToString();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            int ivideodevicecount;
            int iaudiodevicecount;
            string strApp;
            int iFindIndex;
            int i;

           

            strApp = Application.ExecutablePath;

            iFindIndex = strApp.IndexOf("Examples");

            if (iFindIndex != -1)
            {
                TextBox1.Text = strApp.Substring(0, iFindIndex) + "novideo.bmp";
            }



            ivideodevicecount = axVideoChatSender1.GetVideoDeviceCount();

            for (i = 0; i < ivideodevicecount; i++)
            {

                cbovideodevice.Items.Add(axVideoChatSender1.GetVideoDeviceName((short)i));

            }
            if (cbovideodevice.Items.Count > 0)
                cbovideodevice.SelectedIndex = 0;

            iaudiodevicecount = axVideoChatSender1.GetAudioDeviceCount();

            for (i = 0; i < iaudiodevicecount; i++)
                cboaudiodevice.Items.Add(axVideoChatSender1.GetAudioDeviceName((short)i));

            if (cboaudiodevice.Items.Count > 0)
                cboaudiodevice.SelectedIndex = 0;

            cbovideoformat.Items.Add("160x120");
            cbovideoformat.Items.Add("176x144");
            cbovideoformat.Items.Add("320x240");
            cbovideoformat.Items.Add("352x288");
            cbovideoformat.Items.Add("640x480");
            cbovideoformat.Items.Add("1280x720");
            cbovideoformat.Items.Add("1920x1080");
            cbovideoformat.Items.Add("1920x1280");

            cbovideoformat.SelectedIndex = 4;


            cboaudiocomplex.Items.Add("0");
            cboaudiocomplex.Items.Add("1");
            cboaudiocomplex.Items.Add("2");
            cboaudiocomplex.Items.Add("3");
            cboaudiocomplex.SelectedIndex = 0;



            for (i = 0; i <= 10; i++)
            {


                cboaudioquality.Items.Add(i.ToString());

            }
            cboaudioquality.SelectedIndex = 8;

        }

        private void Button1_Click(object sender, EventArgs e)
        {
            axVideoChatSender1.VideoDevice = (short)cbovideodevice.SelectedIndex;

            axVideoChatSender1.AudioDevice = (short)cboaudiodevice.SelectedIndex;


            axVideoChatSender1.VideoFormat = (short)cbovideoformat.SelectedIndex;
            axVideoChatSender1.FrameRate = Convert.ToInt32(txtframerate.Text, 10);
            axVideoChatSender1.VideoBitrate = Convert.ToInt32(txtVideobitrate.Text, 10);
            axVideoChatSender1.AudioComplexity = (short)cboaudiocomplex.SelectedIndex;
            axVideoChatSender1.AudioQuality = (short)cboaudioquality.SelectedIndex;
            axVideoChatSender1.SendAudioStream = chksendaudio.Checked;
            axVideoChatSender1.SendVideoStream = chksendvideo.Checked;

        
            int iResult = axVideoChatSender1.Connect(txtconnectIP.Text, Convert.ToInt32(txtConnectPortNo.Text, 10));

		
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            axVideoChatSender1.Disconnect();
        }

        private void btnstartlisten_Click(object sender, EventArgs e)
        {


            if (RadioAspectRatio1.Checked)
                axVideoChatReceiver1.VideoWindowAspectRatio = 0;
            else if (RadioAspectRatio2.Checked)
                axVideoChatReceiver1.VideoWindowAspectRatio = 1;
            else if (RadioAspectRatio3.Checked)
                axVideoChatReceiver1.VideoWindowAspectRatio = 2;

      
       

            
            axVideoChatReceiver1.ReceiveAudioStream = chkreceiveaudio.Checked;
            axVideoChatReceiver1.ReceiveVideoStream = chkreceivevideo.Checked;
            axVideoChatReceiver1.ReceiverVideoNoVideoImage(TextBox1.Text);

           bool bResult = false;

           bResult = axVideoChatReceiver1.Listen(txtListenIP.Text, Convert.ToInt32(txtConnectPortNo.Text, 10));

           btnstartlisten.Enabled = false;
           btnlistendisconnect.Enabled = true;

        }

        private void btnlistendisconnect_Click(object sender, EventArgs e)
        {
            axVideoChatReceiver1.Disconnect();
            btnstartlisten.Enabled = true;
            btnlistendisconnect.Enabled = false;
        }

        private void btnsnapshot_Click(object sender, EventArgs e)
        {
            SaveFileDialog1.Filter = "JPG File (*.jpg)|*.jpg";

            if (SaveFileDialog1.ShowDialog() == DialogResult.OK)
                axVideoChatReceiver1.SnapShot(SaveFileDialog1.FileName);
    
        }
    }
}

File: vb.net2010/Client/Form1.vb
Public Class Form1

    Private Sub Form1_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load
        Dim iVideoCount As Integer
        Dim iAudioCount As Integer
        Dim strApp As String
        Dim iFIndIndex As Integer

        strApp = Application.ExecutablePath

        iFindIndex = strApp.IndexOf("Examples")

        If iFindIndex <> -1 Then
            strApp = strApp.Substring(0, iFindIndex) + "novideo.bmp"
        End If

        TextBox1.Text = strApp


        iVideoCount = AxVideoChatSender1.GetVideoDeviceCount

        For i = 0 To iVideoCount - 1
            cbovideodevice.Items.Add(AxVideoChatSender1.GetVideoDeviceName(i))
        Next

        If cbovideodevice.Items.Count > 0 Then
            cbovideodevice.SelectedIndex = 0
        End If

        iAudioCount = AxVideoChatSender1.GetAudioDeviceCount

        For i = 0 To iAudioCount - 1
            cboaudiodevice.Items.Add(AxVideoChatSender1.GetAudioDeviceName(i))
        Next

        If cboaudiodevice.Items.Count > 0 Then
            cboaudiodevice.SelectedIndex = 0
        End If

        cbovideoformat.Items.Add("160x120")
        cbovideoformat.Items.Add("176x144")
        cbovideoformat.Items.Add("320x240")
        cbovideoformat.Items.Add("352x288")
        cbovideoformat.Items.Add("640x480")
        cbovideoformat.Items.Add("1280x720")
        cbovideoformat.Items.Add("1920x1080")
        cbovideoformat.Items.Add("1920x1280")


        cbovideoformat.SelectedIndex = 4



        cboaudiocomplex.Items.Add("0")
        cboaudiocomplex.Items.Add("1")
        cboaudiocomplex.Items.Add("2")
        cboaudiocomplex.Items.Add("3")
        cboaudiocomplex.SelectedIndex = 0


        For i = 0 To 10
            cboaudioquality.Items.Add(i.ToString())

        Next


        cboaudioquality.SelectedIndex = 8
    End Sub

    Private Sub Button1_Click(sender As System.Object, e As System.EventArgs) Handles Button1.Click
        AxVideoChatSender1.VideoDevice = cbovideodevice.SelectedIndex
        AxVideoChatSender1.AudioDevice = cboaudiodevice.SelectedIndex
        AxVideoChatSender1.VideoFormat = cbovideoformat.SelectedIndex
        AxVideoChatSender1.FrameRate = txtframerate.Text
        AxVideoChatSender1.VideoBitrate = Convert.ToInt32(txtVideobitrate.Text)
        AxVideoChatSender1.AudioComplexity = cboaudiocomplex.SelectedIndex
        AxVideoChatSender1.AudioQuality = cboaudioquality.SelectedIndex
        AxVideoChatSender1.SendAudioStream = chksendaudio.Checked
        AxVideoChatSender1.SendVideoStream = chksendvideo.Checked


        Dim iresult As Integer
        iresult = AxVideoChatSender1.Connect(txtconnectIP.Text, Convert.ToInt32(txtConnectPortNo.Text))

    End Sub

    Private Sub AxVideoChatSender1_SendStream(sender As System.Object, e As AxVideoChatSenderLib._DvideochatEvents_SendStreamEvent) Handles AxVideoChatSender1.SendStream
        lblvideobitrate.Text = e.fVideoBitrate
        lblaudiobitrate.Text = e.fAudioBitrate
    End Sub

  
    Private Sub btnstartlisten_Click(sender As System.Object, e As System.EventArgs) Handles btnstartlisten.Click
        Dim bResult As Boolean
        AxVideoChatReceiver1.ReceiveAudioStream = chkreceiveaudio.Checked
        AxVideoChatReceiver1.ReceiveVideoStream = chkreceivevideo.Checked
        AxVideoChatReceiver1.ReceiverVideoNoVideoImage(TextBox1.Text)

        If RadioAspectRatio1.Checked Then
            AxVideoChatReceiver1.VideoWindowAspectRatio = 0
        ElseIf RadioAspectRatio2.Checked Then

            AxVideoChatReceiver1.VideoWindowAspectRatio = 1
        ElseIf RadioAspectRatio3.Checked Then
            AxVideoChatReceiver1.VideoWindowAspectRatio = 2
        End If



        bResult = AxVideoChatReceiver1.Listen(txtListenIP.Text, Convert.ToInt32(txtlistenportno.Text, 10))

        btnstartlisten.Enabled = False
        btnlistendisconnect.Enabled = True
    End Sub

    Private Sub btnlistendisconnect_Click(sender As System.Object, e As System.EventArgs) Handles btnlistendisconnect.Click
        AxVideoChatReceiver1.Disconnect()
        btnstartlisten.Enabled = True
        btnlistendisconnect.Enabled = False

    End Sub

    Private Sub btnsnapshot_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnsnapshot.Click
        SaveFileDialog1.Filter = "JPG File (*.jpg)|*.jpg"
        If SaveFileDialog1.ShowDialog() = DialogResult.OK Then
            AxVideoChatReceiver1.SnapShot(SaveFileDialog1.FileName)
        End If
    End Sub

    Private Sub Button2_Click(sender As System.Object, e As System.EventArgs) Handles Button2.Click
        AxVideoChatSender1.Disconnect()
    End Sub
End Class
